FluentTc/Locators/GitVCSRootBuilder.cs

```diff
@@ -0,0 +1,270 @@
+                }
+            );
+            return this;
+        }
+
+        public IGitVCSRootBuilder Password(string value)
+        {
+            m_VCSRoot.Properties.Property.Add(
+                new Property()
+                {
+                    Name = "secure:password",
+                    Value = value
+                }
+            );
+            return this;
+        }
+
+        public IGitVCSRootBuilder UseAlternates(bool value)
+        {
+            m_VCSRoot.Properties.Property.Add(
+                new Property()
+                {
+                    Name = "useAlternates",
+                    Value = value.ToString(CultureInfo.InvariantCulture).ToLower()
+                }
+            );
+            return this;
+        }
+
+        public IGitVCSRootBuilder Url(Uri value)
+        {
+            m_VCSRoot.Properties.Property.Add(
+                new Property()
+                {
+                    Name = "url",
+                    Value = value.ToString()
+                }
+            );
+            return this;
+        }
+
+        public IGitVCSRootBuilder BranchSpec(string value)
+        {
+            m_VCSRoot.Properties.Property.Add(
+                new Property()
+                {
+                    Name = "teamcity:branchSpec",
+                    Value = value
+                }
+            );
+            return this;
+        }
+
+        public IGitVCSRootBuilder SubModuleCheckout(bool value)
+        {
+            m_VCSRoot.Properties.Property.Add(
+                new Property()
+                {
+                    Name = "teamcity:branchSpec",
+                    Value = (value ? "true" : "false")
+                }
+            );
+            return this;
+        }
+
+        public IGitVCSRootBuilder Project(Project value)
+        {
+            m_VCSRoot.Project = value;
+            return this;
+        }
+
+        public VcsRoot GetVCSRoot()
+        {
+            return m_VCSRoot;
+        }
+
+        private string FromPascalToCapitalizedCase(string value)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            int charCount = 0;
+            foreach (char c in value)
+            {
+                charCount++;
+                if (char.IsUpper(c) && charCount != value.Length && charCount != 1)
+                {
+                    sb.Append("_");
+                }
+                sb.Append(char.ToUpper(c));
+            }
+            return sb.ToString();
+        }
+
+        private string FromPascalToCamelCase(string value)
+        {
+            var len = value.Length;
+            if (len > 0)
+            {
+                var sb = new StringBuilder();
+                sb.Append(char.ToLower(value[0]));
+                if (len > 1)
+                    sb.Append(value.Substring(1, len - 1));
+                return sb.ToString();
+            }
+            else
+                return "";
+        }
```

Comment:
Let's call the folder Extensions with separate files for different extension classes: StringExtensions.cs and others